substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -672,11 +674,28 @@ pub mod pallet {
 
             log::info!(
                 "{:?} contracts to bill at block {:?}",
-                contracts,
+                contract_ids,
                 block_number
             );
 
-            for contract_id in contracts {
+            for contract_id in contract_ids {
+                if let Some(c) = Contracts::<T>::get(contract_id) {
+                    if let types::ContractData::NodeContract(node_contract) = c.contract_type {
+                        let bill_ip = node_contract.public_ips > 0;
+                        let bill_cu_su = NodeContractResources::<T>::contains_key(contract_id)
+                            && !NodeContractResources::<T>::get(contract_id).used.is_empty();
+                        let bill_nu =
+                            ContractBillingInformationByID::<T>::contains_key(contract_id)
+                                && ContractBillingInformationByID::<T>::get(contract_id)
```

Comment:
why `contains_key` and `get` afterwards ? I think `get` suffices because of the `ValueQuery` storage definition (default when not present)

Reply:
same idea here